eng/pipelines/scripts/Compare-CurrentToCodegeneration.ps1

```diff
@@ -0,0 +1,58 @@
+<#
+.SYNOPSIS
+Invokes all 'Update-Codegeneration.ps1' scripts found within the specified directory and compares it against current
+code.
+
+.DESCRIPTION
+Invokes all 'Update-Codegeneration.ps1' scripts found within the specified directory and compares it against current
+code.
+
+If the regenerated code is different than the current code this will tell the differences, the files the differences
+are in, and exit with a failure status.
+
+.PARAMETER Directory
+The directory that will be searched for 'Update-Codegeneration.ps1' scripts. The default is the root directory of the
+Azure SDK for Java repository. CI jobs should use the 'ServiceDirectory', such as /sdk/storage.
+#>
+
+param(
+  [Parameter(Mandatory = $false)]
+  [string]$Directory
+)
+
+$path = ""
+if ($Directory) {
+  $path = $Directory
+}
+
+Write-Host "
+
+===================================
+Invoking Autorest code regeneration
+===================================
+
+"
+
+foreach ($script in (Get-ChildItem -Path $path -Filter "Update-Codegeneration.ps1" -Recurse)) {
+  Invoke-Expression $script.FullName
```

Comment:
Do you need to call all the script independently? Is that because you expect them to have custom options or something that they configured?

Reply:
Correct, each script will need to be called independently as they can have different setups. Right now, in this PR the only SDKs I did were Storage which are relatively straight forward in terms of code generation, there are other SDKs such as Cognitive Search where there are multiple `autorest` calls that need to be made to generate the SDK.